core/src/main/scala/better/files/File.scala

```diff
@@ -722,24 +722,31 @@ class File private(val path: Path) {
   /**
     * @return true if file is not present or empty directory or 0-bytes file
     */
-  def isEmpty: Boolean = {
-    this match {
-      case File.Type.Directory(children) => children.isEmpty
-      case File.Type.RegularFile(content) => content.isEmpty
-      case _ => notExists
+  def isEmpty(implicit linkOptions: File.LinkOptions = File.LinkOptions.default): Boolean = {
+    if (isDirectory(linkOptions)) {
+      children.isEmpty
+    } else if (isRegularFile(linkOptions))  {
+      toJava.length() == 0
+    } else {
+      notExists(linkOptions)
     }
   }
 
+  def isNotEmpty(implicit linkOptions: File.LinkOptions = File.LinkOptions.default): Boolean = {
+    !isEmpty(linkOptions)
+  }
```

Comment:
If this is a new method in the upcoming release, consider renaming it to `nonEmpty` by analogy with the collections methods `isEmpty`/`nonEmpty`.